### PR TITLE
CLN: dont special-case DatetimeArray indexing

### LINKED ISSUE #36210: API/DEPR: casting bool-indexer to slice in dt64/td64/period
We use special logic for dt64/td64/period dtypes that makes view/copy behavior different from other dtypes:

```
dti = pd.date_range("2016-01-01", periods=4, tz="US/Pacific")
key = np.array([True, True, False, False])

ser1 = pd.Series(dti._data)
ser2 = pd.Series(range(4))

res1 = ser1[key]
res2 = ser2[key]

>>> res1._values._data.base is None
False
>>> res2._values.base is None
True
```

cc @jorisvandenbossche IIRC you advocated not doing this special casing.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #36210
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

cc @jorisvandenbossche

### CODE REVIEW

**Comment by jreback** [comment]
can you add some tests that lock this down (eg from the OP).

I think this would be ok to push for 1.2 with a whatsnew note.

cc @jorisvandenbossche

**Comment by jbrockmendel** [comment]
whatsnew+test+green

**Comment by jbrockmendel** [comment]
any other thoughts here?  after this i can get back to de-duplicating EA code

File: pandas/tests/series/indexing/test_boolean.py [hunk 1]
```diff
@@ -128,3 +128,14 @@ def test_get_set_boolean_different_order(string_series):
     sel = string_series[ordered > 0]
     exp = string_series[string_series > 0]
     tm.assert_series_equal(sel, exp)
+
+
+def test_getitem_boolean_dt64_copies():
+    # GH#36210
+    dti = date_range("2016-01-01", periods=4, tz="US/Pacific")
+    key = np.array([True, True, False, False])
```
Comment (jreback) [inline pandas/tests/series/indexing/test_boolean.py:136]: can you replicate the OP here as well (e.g. add the slice test for this too)

Comment (jbrockmendel) [reply pandas/tests/series/indexing/test_boolean.py:136]: not sure i understand.  this is exactly the example from the OP

Comment (jreback) [reply pandas/tests/series/indexing/test_boolean.py:136]: the OP had 2 example inputs
```
er1 = pd.Series(dti._data)
ser2 = pd.Series(range(4))
```

**Comment by jbrockmendel** [comment]
Test amended + green

**Review by jreback — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks Brock!

**Pull request merged**